uttut/pipeline/ops/base.py

```diff
@@ -0,0 +1,70 @@
+from abc import ABC, abstractmethod
+from typing import Tuple, List, Union
+
+
+class Operator(ABC):
+    """Base class for Ops
+
+    Sub-classes should implement `transform` and `relabel`.
+
+    Attributes:
+        input_type: input type of datum to transform
+        output_type: output type of transformed datum
+    """
+
+    def __init__(self, input_type, output_type):
+        self._input_type = input_type
+        self._output_type = output_type
+        self.reset_bian()
+
+    def reset_bian(self):
+        self.bian = None
+
+    def update_bian(self, bian_group):
+        self.bian = bian_group
+
+    @property
+    def input_type(self):
+        return self._input_type
+
+    @property
+    def output_type(self):
+        return self._output_type
+
+    @abstractmethod
+    def transform(
+            self,
+            input_sequence: Union[str, List[str]],
+            labels: List[int],
+            state: dict = None,
+        ) -> Tuple[Union[str, List[str]], List[int]]:
+        """Transform input and label and Update self.bian
```

Comment:
```suggestion
        """Transform input and label and update self.bian
```

Reply:
update -> "record"?

Reply:
"Calls to transform do not accumulate changes in self.bian. Each new calls effectively resets the state of the object"

Reply:
ae1204c